samples/texture_vertex.html

```diff
@@ -0,0 +1,299 @@
+<!DOCTYPE html>
+
+<html lang="en">
+
+<head>
+    <title>WebGL 2 Samples - texture_vertex</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, user-scalable=no, minimum-scale=1.0, maximum-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <script src="third-party/gl-matrix.js"></script>
+    <script src="third-party/webgl-obj-loader.js"></script>
+    <script src="utility.js"></script>
+</head>
+
+<body>
+    <div id="info">WebGL 2 Samples - texture_vertex</div>
+    <p id="description">This sample demonstrates the use of texture look up operations in a vertex shader for displacement mapping. </p>
+
+    <!-- WebGL 2 shaders -->
+    <script id="vs" type="x-shader/x-vertex">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+
+        uniform mat4 mvMatrix;
+        uniform mat4 pMatrix;
+        uniform sampler2D displacementMap;
+
+        layout(location = 0) in vec3 position;
+        layout(location = 1) in vec2 texcoord;
+        layout(location = 2) in vec3 normal;
+
+        out vec2 st;
+        out vec3 vPosition;
+
+        void main()
+        {
+            st = texcoord;
+            float height = texture(displacementMap, texcoord).b;
+            vec4 displacedPosition = vec4(position, 1.0) + vec4(normal * height, 0.0);
+            vPosition = vec3(mvMatrix * displacedPosition);
+            gl_Position = pMatrix * mvMatrix * displacedPosition;
+        }
+    </script>
+
+    <script id="fs" type="x-shader/x-fragment">
+        #version 300 es
+        precision highp float;
+        precision highp int;
+        precision highp sampler2D;
+
+        uniform sampler2D diffuse;
+
+        in vec2 st;
+        in vec3 vPosition;
+
+        out vec4 color;
+
+        float textureLevel(in sampler2D sampler, in vec2 st)
+        {
+            vec2 size = vec2(textureSize(sampler, 0));
+
+            float levelCount = max(log2(size.x), log2(size.y));
+
+            vec2 dx = dFdx(st * size);
+            vec2 dy = dFdy(st * size);
+            float d = max(dot(dx, dx), dot(dy, dy));
+
+            d = clamp(d, 1.0, pow(2.0, (levelCount - 1.0) * 2.0));
+
+            return 0.5 * log2(d);
+        }
+
+        void main()
+        {
+            vec2 sampleCoord = fract(st.xy);
+            float level = textureLevel(diffuse, st);
+
+            // Compute LOD using gradient
+            color = textureLod(diffuse, st, level);
+
+            // Compute flat normal using gradient
+            vec3 fdx = vec3(dFdx(vPosition.x), dFdx(vPosition.y), dFdx(vPosition.z));
```

Comment:
I _think_ this can be `dFdx(vPosition)`.

See https://www.opengl.org/sdk/docs/man/html/dFdx.xhtml